featurebyte/session/base.py

```diff
@@ -238,6 +262,47 @@ def connection(self) -> Any:
         """
         return self._connection
 
+    async def _cancel_query(self, cursor: Any, query: str) -> bool:
+        """
+        Cancel query
+
+        Parameters
+        ----------
+        cursor : Any
+            The connection cursor
+        query : str
+            SQL query
+
+        Returns
+        -------
+        bool
+            True if query was cancelled, False otherwise
+        """
+        try:
+            cursor.cancel()
+            return True
+        except AttributeError:
+            return False
```

Comment:
It seems almost all sessions have to override this implementation. In which session type would this default implementation work?

Reply:
There is SQLiteSession. Would make sense to make this an abstractmethod and add an implementation there?

Reply:
I think would be good to make this an abstractmethod since most likely it needs to have a custom implementation.